external/storm-hdfs-oci/src/main/java/org/apache/storm/container/oci/LocalOrHdfsImageTagToManifestPlugin.java

```diff
@@ -0,0 +1,294 @@
+                ret = true;
+            }
+        }
+        return ret;
+    }
+
+    private BufferedReader getLocalImageToHashReader() throws IOException {
+        if (localImageTagToHashFile == null) {
+            LOG.debug("Did not load local image to hash file, file is null");
+            return null;
+        }
+
+        File imageTagToHashFile = new File(localImageTagToHashFile);
+        if (!imageTagToHashFile.exists()) {
+            LOG.warn("Did not load local image to hash file, file doesn't exist");
+            return null;
+        }
+
+        long newLocalModTime = imageTagToHashFile.lastModified();
+        if (newLocalModTime == localModTime) {
+            LOG.debug("Did not load local image to hash file, file is unmodified");
+            return null;
+        }
+        localModTime = newLocalModTime;
+
+        return new BufferedReader(new FileReader(imageTagToHashFile));
+    }
+
+    private BufferedReader getHdfsImageToHashReader() throws IOException {
+        if (hdfsImageToHashFile == null) {
+            LOG.debug("Did not load hdfs image to hash file, file is null");
+            return null;
+        }
+
+        Path imageToHash = new Path(hdfsImageToHashFile);
+        FileSystem fs = imageToHash.getFileSystem(new Configuration());
+        if (!fs.exists(imageToHash)) {
+            String message = "Could not load hdfs image to hash file, " + hdfsImageToHashFile + " doesn't exist";
+            LOG.error(message);
+            throw new IOException(message);
+        }
+
+        long newHdfsModTime = fs.getFileStatus(imageToHash).getModificationTime();
+        if (newHdfsModTime == hdfsModTime) {
+            LOG.debug("Did not load hdfs image to hash file, file is unmodified");
+            return null;
+        }
+        hdfsModTime = newHdfsModTime;
+
+        return new BufferedReader(new InputStreamReader(fs.open(imageToHash)));
+    }
+
+    // You may specify multiple tags per hash all on the same line.
+    // Comments are allowed using #. Anything after this character will not
+    // be read
+    // Example file:
+    // foo/bar:current,fizz/gig:latest:123456789
+    // #this/line:wont,be:parsed:2378590895
+    //
+    // This will map both foo/bar:current and fizz/gig:latest to 123456789
+    private static Map<String, String> readImageToHashFile(BufferedReader br) throws IOException {
+        if (br == null) {
+            return null;
+        }
+
+        String line;
+        Map<String, String> imageToHashCache = new HashMap<>();
+        while ((line = br.readLine()) != null) {
+            int index;
+            index = line.indexOf("#");
+            if (index == 0) {
+                continue;
+            } else if (index != -1) {
+                line = line.substring(0, index);
+            }
+
+            index = line.lastIndexOf(":");
+            if (index == -1) {
+                LOG.warn("Malformed imageTagToManifest entry: " + line);
```

Comment:
It will be good to know which file or hdfs url this error is in. Same comment for line 214.